### ISSUE TITLE
"Couldn't Establish Connection" message when trying to connect

### ISSUE THREAD

**Issue by ids1024** [owncloud/android#1477]
I am trying to get the app to connect to my owncloud, but get the message "Couldn't Establish Connection".

Here is the adb logcat output:

```logcat
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation: Connection check at http://owncloud.iandouglasscott.com: Socket exception
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation: java.net.ConnectException: failed to connect to owncloud.iandouglasscott.com/2001:19f0:ac00:be6:5400:ff:fe1b:b1b4 (port 443) after 5000ms: isConnected failed: ECONNREFUSED (Connection refused)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at libcore.io.IoBridge.isConnected(IoBridge.java:234)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at libcore.io.IoBridge.connectErrno(IoBridge.java:171)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at libcore.io.IoBridge.connect(IoBridge.java:122)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:183)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:452)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at java.net.Socket.connect(Socket.java:884)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at com.owncloud.android.lib.common.network.AdvancedSslSocketFactory.createSocket(AdvancedSslSocketFactory.java:184)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at org.apache.commons.httpclient.HttpConnection.open(HttpConnection.java:707)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at org.apache.commons.httpclient.MultiThreadedHttpConnectionManager$HttpConnectionAdapter.open(MultiThreadedHttpConnectionManager.java:1361)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at org.apache.commons.httpclient.HttpMethodDirector.executeWithRetry(HttpMethodDirector.java:387)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at org.apache.commons.httpclient.HttpMethodDirector.executeMethod(HttpMethodDirector.java:171)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at org.apache.commons.httpclient.HttpClient.executeMethod(HttpClient.java:397)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at org.apache.commons.httpclient.HttpClient.executeMethod(HttpClient.java:323)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at com.owncloud.android.lib.common.OwnCloudClient.executeMethod(OwnCloudClient.java:222)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at com.owncloud.android.lib.common.OwnCloudClient.executeMethod(OwnCloudClient.java:192)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at com.owncloud.android.lib.resources.status.GetRemoteStatusOperation.tryConnection(GetRemoteStatusOperation.java:106)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at com.owncloud.android.lib.resources.status.GetRemoteStatusOperation.run(GetRemoteStatusOperation.java:202)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at com.owncloud.android.lib.common.operations.RemoteOperation.execute(RemoteOperation.java:136)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at com.owncloud.android.operations.GetServerInfoOperation.run(GetServerInfoOperation.java:81)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at com.owncloud.android.lib.common.operations.RemoteOperation.execute(RemoteOperation.java:136)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at com.owncloud.android.services.OperationsService$ServiceHandler.nextOperation(OperationsService.java:486)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at com.owncloud.android.services.OperationsService$ServiceHandler.handleMessage(OperationsService.java:421)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at android.os.Handler.dispatchMessage(Handler.java:102)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at android.os.Looper.loop(Looper.java:148)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at android.os.HandlerThread.run(HandlerThread.java:61)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation: Caused by: android.system.ErrnoException: isConnected failed: ECONNREFUSED (Connection refused)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      at libcore.io.IoBridge.isConnected(IoBridge.java:223)
02-13 16:10:38.887  4091 11713 E GetRemoteStatusOperation:      ... 24 more
```

As far as I understand, this says it is failing to retrieve https://owncloud.iandouglasscott.com/status.php, but that loads properly in the browser on my phone, so I don't know why this isn't working. It is probably a server-side issue, since using https://demo.owncloud.org as the url works.

Edit: The Linux desktop client seems to be working fine.

**Comment by tobiasKaminsky** [comment]
Are you using some kind of IPv6?
As far as I know this is currently not supported by the app and therefore it fails.
(please correct me if I am wrong!)

**Comment by ids1024** [comment]
My server and wifi network support ipv4 and ipv6, so it defaults to the latter. It wouldn't think that would cause any problem, but if I disable ipv6 with https://play.google.com/store/apps/details?id=de.lennartschoch.disableipv6, it works! That said, it is inconvenient to disable ipv6 on my phone just to fix this...

**Comment by ids1024** [comment]
Oh, it turns out my nginx server wasn't configured to serve over ipv6. Owncloud should probably fallback to ipv4 as mentioned in #616, but that fixes it anyway.

**Comment by tobiasKaminsky** [comment]
Real ipv6 needs to be done sooner or later.
So, as this is "solved" by you, I close it?

**Comment by ids1024** [comment]
Yes, it is solved, other than the fact it isn't falling back to ipv4, which is already addressed in #616.
